### PR TITLE
virttest: add function to get cache memory for given OS

### PR DESCRIPTION
**Pull request by xiagao**
Get cache memory of guest,including windows and linux os.
ID: 1946456
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
@huangyum Could you help review this patch?Thanks.

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -2320,6 +2320,28 @@ def get_free_mem(session, os_type):
...
+        cache = get_mem_info(session, 'Cached') + \
+                get_mem_info(session, 'Buffers')
+        cache = "%s kB" % cache
+    else:
+        cmd_get_cache = 'powershell -command "get-ciminstance -classname'
+        cmd_get_cache += ' win32_perfrawdata_perfos_memory |select CacheBytes"'
+        output = session.cmd_output(cmd_get_cache)
+        cache = "%s" % re.findall("\d+", output)[0]
+    cache = float(normalize_data_size(cache, order_magnitude="M"))
+    return int(cache)
```
Comment (huangyum) [inline virttest/utils_misc.py:2342]: I guess you don't need to convert 'cache' twice.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -2320,6 +2320,28 @@ def get_free_mem(session, os_type):
...
+def get_cache_mem(session, os_type):
+    """
+    Get cache memory for given OS.
+
+    :param session: shell Object.
+    :param os_type: os type (eg. linux or windows)
+
+    :return string: memory cache M-bytes
+    """
+    if os_type != "windows":
```
Comment (huangyum) [inline virttest/utils_misc.py:2332]: Prefer `if os_type == "linux"`.

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -2320,6 +2320,28 @@ def get_free_mem(session, os_type):
...
 
 
+def get_cache_mem(session, os_type):
+    """
+    Get cache memory for given OS.
+
+    :param session: shell Object.
+    :param os_type: os type (eg. linux or windows)
+
+    :return string: memory cache M-bytes
```
Comment (huangyum) [inline virttest/utils_misc.py:2330]: Seems the return is not string.

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -2320,6 +2320,28 @@ def get_free_mem(session, os_type):
...
+    """
+    if os_type != "windows":
+        cache = get_mem_info(session, 'Cached') + \
+                get_mem_info(session, 'Buffers')
+        cache = "%s kB" % cache
+    else:
+        cmd_get_cache = 'powershell -command "get-ciminstance -classname'
+        cmd_get_cache += ' win32_perfrawdata_perfos_memory |select CacheBytes"'
+        output = session.cmd_output(cmd_get_cache)
+        cache = "%s" % re.findall("\d+", output)[0]
```
Comment (huangyum) [inline virttest/utils_misc.py:2340]: cache = re.findall("\d+", output)[0]

**Comment by xiagao** [comment]
@huangyum update.

**Comment by huangyum** [comment]
> @huangyum update.

Forgot to push?

**Comment by xiagao** [comment]
> > @huangyum update.
> 
> Forgot to push?

done.

**Review by huangyum — APPROVED** [review]
ACK

**Review by vivianQizhu — APPROVED** [review]

File: virttest/utils_misc.py [hunk 5]
```diff
@@ -2320,6 +2320,29 @@ def get_free_mem(session, os_type):
...
+    :return: memory cache M-bytes
+    """
+    if os_type == "windows":
+        cmd_get_cache = 'powershell -command "get-ciminstance -classname'
+        cmd_get_cache += ' win32_perfrawdata_perfos_memory |select CacheBytes"'
+        output = session.cmd_output(cmd_get_cache)
+        cache = re.findall("\d+", output)[0]
+    else:
+        cache = get_mem_info(session, 'Cached') + \
+                get_mem_info(session, 'Buffers')
```
Comment (vivianQizhu) [inline virttest/utils_misc.py:2339]: Wait, better avoid `\`:
```suggestion
        cache = (get_mem_info(session, 'Cached') +
                 get_mem_info(session, 'Buffers'))
```

Comment (xiagao) [reply virttest/utils_misc.py]: update, could you have a check?Thanks.

**Pull request merged**